components/encoder/ams/ams_as5048.go

```diff
@@ -188,25 +184,18 @@ func (enc *Encoder) Reconfigure(
 	return nil
 }
 
-func (enc *Encoder) startPositionLoop(ctx context.Context) error {
-	if err := enc.ResetPosition(ctx, map[string]interface{}{}); err != nil {
-		return err
-	}
-	enc.activeBackgroundWorkers.Add(1)
-	utils.ManagedGo(func() {
-		for {
-			if enc.cancelCtx.Err() != nil {
-				return
-			}
-			if err := enc.updatePosition(enc.cancelCtx); err != nil {
-				enc.logger.CErrorf(ctx,
-					"error in position loop (skipping update): %s", err.Error(),
-				)
-			}
-			time.Sleep(time.Duration(waitTimeNano))
+func (enc *Encoder) positionLoop(cancelCtx context.Context) {
```

Comment:
Here's what happened with this function:
- The first 3 lines about `enc.ResetPosition` got inlined into `makeAS5048Encoder`
- The rest of it was just spinning up the background goroutine. Instead, this function is now the implementation of the goroutine, and it gets spun up at the end of `makeAS5048Encoder`, just after the new call to `ResetPosition`. This involved unindenting the majority of the function a level; the diff looks simpler if you ignore whitespace changes.
- Changed the name from `startPositionLoop` to just `positionLoop`, since it's now started elsewhere.